src/common/utils/container.hpp

```diff
@@ -18,8 +18,39 @@ namespace utils
         }
     };
 
+    struct insensitive_string_hash
+    {
+        using is_transparent = void;
+
+        size_t operator()(const std::string_view str) const
+        {
+            size_t hash = 0;
+            constexpr std::hash<int> hasher{};
+            for (unsigned char c : str)
+            {
+                hash ^= hasher(std::tolower(c)) + 0x9e3779b9 + (hash << 6) + (hash >> 2);
```

Comment:
you can include "string.hpp" and do this:

```suggestion
            for (const char c : str)
            {
                hash ^= hasher(utils::string::char_to_lower(c)) + 0x9e3779b9 + (hash << 6) + (hash >> 2);
```

that automatically takes care of the int/unsigned char weirdness std::tolower brings